packages/cli/src/workflows/workflows.controller.ts

```diff
@@ -112,14 +114,29 @@ export class WorkflowsController {
 		await Db.transaction(async (transactionManager) => {
 			savedWorkflow = await transactionManager.save<WorkflowEntity>(newWorkflow);
 
-			const project = await this.projectRepository.getPersonalProjectForUserOrFail(req.user.id);
+			const { projectId } = req.body;
+			const project =
+				projectId === undefined
+					? // TODO: pass the transaction manager optionally?
+					  await this.projectRepository.getPersonalProjectForUser(req.user.id)
+					: await this.projectService.getProjectWithScope(req.user, projectId, 'workflow:create');
 
-			const newSharedWorkflow = new SharedWorkflow();
+			if (typeof projectId === 'string' && project === null) {
```

Comment:
nit: Might a bit overkill, but if the personal project is not found (for whatever reason) can we throw a specific error and log it to sentry? I hope this doesn't happen ever, but it would be great to know if it ever does.

Reply:
Sure, is this centralized, like a middleware that catches errors and logs some of them to sentry?

Reply:
Done it here:
https://github.com/n8n-io/n8n/pull/8662/files#diff-6c9bd8e6d42abb8d96a8ed9ea71c48883b87ef6bd162af1aae7353ffff554a4aR142

The line you commented on will return a `BadRequestError` because the user passed a `projectId` but did not have the permissions to create a workflow in it.